pkg/controller/certificates/internal/secretsmanager/secret_test.go

```diff
@@ -71,6 +74,15 @@ func TestSecretsManager(t *testing.T) {
 			"template": "label",
 		}),
 	)
+	baseCertWithAdditionalKeyFormats := gen.CertificateFrom(baseCertBundle.Certificate,
+		gen.SetCertificateAdditionalKeyFormats([]cmapi.AdditionalKeyFormat{
+			{Type: "DER"},
+			{Type: "PEM"},
+		}),
+		gen.EnableFeatureGate(feature.ExperimentalCertificatePrivateKeyAdditionalFormats),
```

Comment:
This will leak across tests. Please unset this feature gate in a testing `t.Cleanup` function.

Reply:
Is it ok if I disable the feature in the CheckFn of the test builder ?

Reply:
Please remove the `gen.EnableFeatureGate` function (as it isn't needed from what I can tell?) and instead just use the feature util package as documented: https://github.com/jetstack/cert-manager/blob/4afe2f00a8142d0bd2134da408a811cb70e100a6/pkg/util/feature/feature_gate.go#L27

Should be:

```
defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ExperimentalCertificatePrivateKeyAdditionalFormats, true)()
```

Reply:
The `defer` will take care of unsetting this after the test :)

Reply:
done :heavy_check_mark: